chain/processor.go

```diff
@@ -100,6 +110,7 @@ func (b *StatelessBlock) Execute(
 					toCache[k] = &fetchData{v, true, numChunks}
```

Comment:
We also update the cache with an `RLock` (which probably is unexpected behavior)